settings.gradle

```diff
@@ -123,15 +39,177 @@ include 'lowcodelanguage'
+        //@formatter:on
+    }
+
+    // NOTE: We are adding dependencies to projects instead of through settings, so we don't have to apply forgedev in settings.
+    // Applying forgedev in settings will break IDE linting support, which makes it very hard to work with.
+    // We can reconsider this if/when switching to Kotlin DSL, or if JetBrains gets their shit together.
+    project.repositories {
+        mavenCentral()
+        maven { url = 'https://libraries.minecraft.net/' }
+
+        project.pluginManager.withPlugin('net.minecraftforge.gradleutils') {
+            maven project.gradleutils.forgeMaven
+        }
+
+        project.pluginManager.withPlugin('net.minecraftforge.forgedev') {
+            maven project.forgedev.mavenizer
+        }
+
+        mavenLocal()
+    }
+}
+
+dependencyResolutionManagement.versionCatalogs {
+    register('bootLibs') {
+        description = 'Important version numbers that are externally declared for easy modification.'
+
+        // Contains Java, Minecraft, and MCP versions
+        from(files('minecraft.versions.toml'))
+    }
 
-include ':forge'
-project(":forge").projectDir = file("projects/forge")
-project(':forge').buildFileName = '../../build_forge.gradle'
+    register('buildLibs') {
+        description = 'Libraries used solely during the build and publish process.'
 
-if (false && !System.env.TEAMCITY_VERSION) {
-    include ':clean'
-    project(':clean').projectDir = file('projects/clean')
-    project(':clean').buildFileName = '../../build_clean.gradle'
+        library 'binarypatcher',  'net.minecraftforge', 'binarypatcher'         version '1.2.0'
+        library 'installertools', 'net.minecraftforge', 'installertools'        version '1.4.3'
+        library 'fart',           'net.minecraftforge', 'ForgeAutoRenamingTool' version '1.0.6'
+        library 'srg2source',     'net.minecraftforge', 'Srg2Source'            version '8.1.1'
+    }
+
+    //@formatter:off
+    register('libs') {
+        description = 'Libraries used by Forge and its subprojects.'
+
+        plugin 'licenser',    'net.minecraftforge.licenser'        version '1.2.0'
+        plugin 'versions',    'com.github.ben-manes.versions'      version '0.53.0'
+        plugin 'gradleutils', 'net.minecraftforge.gradleutils'     version '3.3.35'
+        plugin 'gitversion',  'net.minecraftforge.gitversion'      version '3.1.7'
+        plugin 'changelog',   'net.minecraftforge.changelog'       version '3.2.1'
+        plugin 'download',    'de.undercouch.download'             version '5.6.0'
+        plugin 'jarsigner',   'net.minecraftforge.gradlejarsigner' version '1.2.0'
+        plugin 'apt',         'com.diffplug.eclipse.apt'           version '4.4.1'
+
+        // TODO [Forge][Buildscript We should not be needing to manually add transitive dependencies
+        //      If we need to make strict versions for dependencies, we should do so using dependency constraints
```

Comment:
We need to manually set transitive dependencies so that the installer only includes the things that we need.
We dont want random things like guava's fake annotation library.